src/v2/trades.ts

```diff
@@ -0,0 +1,63 @@
+import { getAddress } from '@ethersproject/address'
+import { NowRequest, NowResponse } from '@now/node'
+import { BigNumber } from '@uniswap/sdk'
+
+import { getSwaps } from './_shared'
+import { return200, return400, return500 } from '../utils'
+
+export default async function(req: NowRequest, res: NowResponse): Promise<void> {
+  if (
+    !req.query.pair ||
+    typeof req.query.pair !== 'string' ||
+    !/^0x[0-9a-fA-F]{40}_0x[0-9a-fA-F]{40}$/.test(req.query.pair)
+  ) {
+    return400(res, 'Invalid pair identifier: must be of format tokenAddress_tokenAddress')
+    return
+  }
+
+  const [tokenA, tokenB] = req.query.pair.split('_')
+  let idA: string, idB: string
+  try {
+    ;[idA, idB] = [getAddress(tokenA), getAddress(tokenB)]
+  } catch (error) {
+    return400(res)
+    return
+  }
+
+  try {
+    const swaps = await getSwaps(idA, idB)
+
+    return200(
+      res,
+      swaps.map(swap => {
+        const aOut = swap.amountAOut !== '0'
+        const bOut = swap.amountBOut !== '0'
+        // a is the base
+        // swapping out the base === buying
+        const type = aOut && bOut ? 'flash' : aOut ? 'buy' : bOut ? 'sell' : 'unknown'
```

Comment:
this is a bit funky. i think it's basically fair to characterize `aOut && bOut` as a flash (although it technically doesn't have to be), but for just `aOut` it's equally possible that it's a swap buy or a flash in either direction, because we don't know if they 1) swapped `b` for `a` (swap buy), 2) flashed `a` and returned `b` (flash buy), or 3) flashed `a` and returned `a` (flash "buy" + "sell").

also sort of dumb but the contract disallows `aOut === 0` and `bOut === 0` so `'unknown'` should maybe be returned after a `??` to indicate this, and we could potentially even add a log to that branch, because if it ever happens it's a bug in the subgraph.